native/dlarfg.go

```diff
@@ -0,0 +1,64 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"math"
+
+	"github.com/gonum/blas/blas64"
+)
+
+// Dlarfg generates an elementary reflector for a household matrix. It creates
+// a real elementary reflector of order n such that
+//  H * (alpha) = (beta)
+//      (    x)   (   0)
+//  H^T * H = I
+// H is represented in the form
+//  H = 1 - tau * (1; v) * (1 v^T)
+// where tau is a real scalar.
+//
+// On entry, x contains the vector x, on exit it contains v.
+//
+// On entry, c contains alpha. On exit it contains beta.
+func (impl Implementation) Dlarfg(n int, c *float64, x []float64, incX int) (tau float64) {
```

Comment:
Yea, I don't know what the right balance is in terms of API changes. I'll change it to not use a pointer.